awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/OtlpAwsSpanExporter.java

```diff
@@ -0,0 +1,163 @@
+
+import io.opentelemetry.exporter.internal.otlp.traces.TraceRequestMarshaler;
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter;
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporterBuilder;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.common.export.MemoryMode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.net.URI;
+import java.util.*;
+import java.util.function.Supplier;
+import javax.annotation.concurrent.Immutable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.http.auth.aws.signer.AwsV4HttpSigner;
+import software.amazon.awssdk.http.auth.spi.signer.SignedRequest;
+
+/**
+ * This exporter extends the functionality of the OtlpHttpSpanExporter to allow spans to be exported
+ * to the XRay OTLP endpoint https://xray.[AWSRegion].amazonaws.com/v1/traces. Utilizes the AWSSDK
+ * library to sign and directly inject SigV4 Authentication to the exported request's headers. <a
+ * href="https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-OTLPEndpoint.html">...</a>
+ */
+@Immutable
+public class OtlpAwsSpanExporter implements SpanExporter {
+  private static final String SERVICE_NAME = "xray";
+  private static final Logger logger = LoggerFactory.getLogger(OtlpAwsSpanExporter.class);
+
+  private final OtlpHttpSpanExporterBuilder parentExporterBuilder;
+  private final OtlpHttpSpanExporter parentExporter;
+  private final String awsRegion;
+  private final String endpoint;
+  private Collection<SpanData> spanData;
+
+  public OtlpAwsSpanExporter(String endpoint) {
+    this(null, endpoint);
+  }
+
+  public OtlpAwsSpanExporter(OtlpHttpSpanExporter parentExporter, String endpoint) {
+    OtlpHttpSpanExporterBuilder defaultExporterBuilder =
+        OtlpHttpSpanExporter.builder()
+            .setMemoryMode(MemoryMode.IMMUTABLE_DATA)
+            .setEndpoint(endpoint)
+            .setHeaders(new SigV4AuthHeaderSupplier());
+
+    this.parentExporterBuilder =
+        parentExporter == null
+            ? defaultExporterBuilder
+            : parentExporter.toBuilder()
+                .setMemoryMode(MemoryMode.IMMUTABLE_DATA)
+                .setEndpoint(endpoint)
+                .setHeaders(new SigV4AuthHeaderSupplier());
+
+    this.parentExporter = this.parentExporterBuilder.build();
+
+    this.awsRegion = endpoint.split("\\.")[1];
+    this.endpoint = endpoint;
+    this.spanData = new ArrayList<>();
+  }
```

Comment:
Shouldn't create the regular builder unless it's needed. Feel free to use `else` instead of `return` to ensure only one of those paths runs.
```suggestion
  public OtlpAwsSpanExporter(OtlpHttpSpanExporter parentExporter, String endpoint) {
    this.awsRegion = endpoint.split("\\.")[1];
    this.endpoint = endpoint;
    this.spanData = new ArrayList<>();
    
    if (parentExporter == null) {
        this.parentExporter =
            OtlpHttpSpanExporter.builder()
                .setMemoryMode(MemoryMode.IMMUTABLE_DATA)
                .setEndpoint(endpoint)
                .setHeaders(new SigV4AuthHeaderSupplier());
        return;
    }
    this.parentExporterBuilder =
        parentExporter.toBuilder()
            .setMemoryMode(MemoryMode.IMMUTABLE_DATA)
            .setEndpoint(endpoint)
            .setHeaders(new SigV4AuthHeaderSupplier());
  }
```